### PR TITLE
MM-T612 Regenerate token

### PR DESCRIPTION
**Pull request by MikeworX**
**_Summary_**

This PR validates that the out-going web-hook's token is indeed regenerated/changed when the user clicks the Regenerate Token link on the Outgoing Webhook's integrations page. This test:

- creates a test team, channel, user, 
- creates an outgoing webhook integration with a trigger word
- isolates the token and stores in a variable
- posts the trigger word to force a bot message from the integration that contains the token value
- validates that the token value is present 
- returns to the integrations page and regenerates the token, also storing it in a variable
- posts the trigger word again, and validating that the second token appears in the message, not the first token

Note: this test requires the webhook server to be running (`npm run start:webhook`)

Screen Shot 2020-10-15 at 4 39 05 PM

**_Links_**

[vercel](https://automation-test-cases.vercel.app/test/MM-T612)
[TM4J](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T612)

### CODE REVIEW

File: e2e/webhook_serve.js [hunk 1]
```diff
@@ -161,3 +162,47 @@ function sendSysadminResponse(message, channelId) {
...
+- response_type: "${responseType}"
+- type: ""
+- username: "${username}"
+- icon_url: "${iconUrl}"
+\`\`\`
+`;
+}
+
+/**
+ * @route "POST /post_outgoing_webhook?override_username=={username}&override_icon_url={iconUrl}&response_type={comment}"
```
Comment (saturninoabril) [inline e2e/webhook_serve.js:185]:
```suggestion
 * @route "POST /post_outgoing_webhook?override_username={username}&override_icon_url={iconUrl}&response_type={comment}"
```

File: e2e/webhook_serve.js [hunk 2]
```diff
@@ -161,3 +162,47 @@ function sendSysadminResponse(message, channelId) {
...
+ * @query override_username - the user name that overrides the user name defined by the outgoing webhook
+ * @query override_icon_url= the user icon url that overrides the user icon url defined by the outgoing webhook
```
Comment (saturninoabril) [inline e2e/webhook_serve.js:187]:
```suggestion
 * @query override_icon_url - the user icon url that overrides the user icon url defined by the outgoing webhook
```

File: e2e/webhook_serve.js [hunk 3]
```diff
@@ -161,3 +162,47 @@ function sendSysadminResponse(message, channelId) {
...
+ * @query response_type= -
```
Comment (saturninoabril) [inline e2e/webhook_serve.js:188]:
```suggestion
 * @query response_type - "in_channel" (default) or "comment"
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @MikeworX! LGTM, except for few requests.

**Comment by josephbaylon** [comment]
@MikeworX Please fix conflicts. Thanks

**Comment by MikeworX** [comment]
@josephbaylon Thanks! I was just consulting Saturn to insure I completed it correctly.

**Review by saturninoabril — APPROVED** [review]
Thanks @MikeworX! LGTM.

**Review by josephbaylon — APPROVED** [review]
LGTM. Test passed. Thanks @MikeworX !!

**Pull request merged**